File: Part 2 - Regression/Section 4 - Simple Linear Regression/simple_linear_regression.py

# DATA PREPROCESSING START
#
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
from sklearn.cross_validation import train_test_split
# from sklearn.preprocessing import StandardScaler

dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, -1].values

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3, random_state = 0)
#
# DATA PREPROCESSING END

# Simple Linear Regression

# Import LinearRegression class from the sklearn library
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()

# Fit regressor model to training data
regressor.fit(X_train, y_train)

# Predict target values from the test set
y_pred = regressor.predict(X_test)

# Visualize training set results
# Generate scatter plot for training data (colored red)
plt.scatter(X_train, y_train, c="red")
# Plot the regression line (colored blue)
plt.plot(X_train, regressor.predict(X_train), c="blue")
# Add Title and Axis Labels
plt.title("Salary vs Years Experience (Training Set)")
plt.xlabel("Years of Experience")
plt.ylabel("Salary ($)")
plt.show()

# Visualize test set results
# Generate scatter plot for test data (colored green)
plt.scatter(X_test, y_test, c="green")
# Plot the regression line (colored blue)
plt.plot(X_train, regressor.predict(X_train), c="blue")
# Add Title and Axis Labels
plt.title("Salary vs Years Experience (Test Set)")
plt.xlabel("Years of Experience")
plt.ylabel("Salary ($)")
plt.show()

File: Part 2 - Regression/Section 4 - Simple Linear Regression/simple_linear_regression.R
# DATA PREPROCESSING START
#
dataset <- read.csv('Salary_Data.csv')
library(caTools)
set.seed(423)
split <- sample.split(dataset$Salary, SplitRatio = 2/3)
training_set <- subset(dataset, split == TRUE)
test_set <- subset(dataset, split == FALSE)
#
# DATA PREPROCESSING END

# Simple Linear Regression
# Fit Simple Linear Regression to Training Set
regressor <- lm(formula = Salary ~ YearsExperience,
               data = training_set)

# Predict Test Set Results
y_pred <- predict(regressor, newdata = test_set)

# Visualize Simple Linear Regression Model
# You need the package "ggplot2" for this.
library(ggplot2)

# Create a new ggplot
# Generate scatter plot for training data and add it to the ggplot
# visualize regression line and add it to the ggplot
# Add Titles and Axis Labels
ggplot() + 
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
             color = "red") +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)), 
            color = "blue") + 
  ggtitle("Salary vs Years Experience (Training Set)") +
  xlab("Years of Experience") + 
  ylab("Salary ($)")

# Visualize Regression Model vs Test Data
ggplot() + 
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary),
             color = "green") +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)), 
            color = "blue") + 
  ggtitle("Salary vs Years Experience (Test Set)") +
  xlab("Years of Experience") + 
  ylab("Salary ($)")